addon/linux_esdm_es/esdm_es_irq.c

```diff
@@ -112,58 +121,100 @@ static u32 esdm_irq_avail_pool_size(void)
 /* Return entropy of unused IRQs present in all per-CPU pools. */
 static u32 esdm_irq_avail_entropy(u32 __unused)
 {
+	u32 block_factor = 1;
 	u32 irq = 0;
+	u32 ent = 0;
 	int cpu;
 
 	/* Only deliver entropy when SP800-90B self test is completed */
 	if (!esdm_sp80090b_startup_complete_es(esdm_int_es_irq))
 		return 0;
 
 	for_each_online_cpu (cpu) {
-		irq += atomic_read_u32(per_cpu_ptr(&esdm_irq_array_irqs, cpu));
+		irq += min_t(u32, ESDM_DATA_NUM_VALUES,
+			     atomic_read_u32(per_cpu_ptr(&esdm_irq_array_irqs, cpu)));
 	}
 
 	/* Consider oversampling rate */
-	return esdm_reduce_by_osr(
+	ent = esdm_reduce_by_osr(
 		esdm_data_to_entropy(irq, esdm_irq_entropy_bits));
+
+	if (fips_enabled) {
+		block_factor++;
```

Comment:
What is this purpose?

Reply:
Will be changed. Purpose: With oversampling in user-space for initial DRBG seeding with e.g. 384 bit, multiple DRBG extractions are needed in kernel space. Both require additional 64 Bit entropy. Therefore count number of additional extractions and reduce entropy estimation at least by them, to always allow at least a single 384 bit extraction from ES or no extraction at all.